src/Utils/StrictPlus.php

```diff
@@ -0,0 +1,232 @@
+			return $val;
+
+		} elseif (isset($methods['set' . $uname])) { // property getter
+			throw new MemberAccessException("Cannot read a write-only property $class::\$$name.");
+
+		} else {
+			ObjectMixin::strictGet($class, $name);
+		}
+	}
+
+
+	/**
+	 * @return void
+	 * @throws MemberAccessException if the property is not defined or is read-only
+	 */
+	public function __set($name, $value)
+	{
+		$class = get_class($this);
+		$uname = ucfirst($name);
+
+		if (ObjectMixin::hasProperty($class, $name)) { // unsetted property
+			$this->$name = $value;
+
+		} elseif ($prop = ObjectMixin::getMagicProperty($class, $name)) { // property setter
+			if (!($prop & 0b1000)) {
+				throw new MemberAccessException("Cannot write to a read-only property $class::\$$name.");
+			}
+			$this->{'set' . $name}($value);
+
+		} elseif ($name === '') {
+			throw new MemberAccessException("Cannot write to a class '$class' property without name.");
+
+		} elseif (($methods = & ObjectMixin::getMethods($class)) && isset($methods[$m = 'set' . $uname])) { // old property setter
+			trigger_error("Missing annotation @property for $class::\$$name used" . ObjectMixin::getSource(), E_USER_DEPRECATED);
+			$this->$m($value);
+
+		} elseif (isset($methods['get' . $uname]) || isset($methods['is' . $uname])) { // property setter
+			throw new MemberAccessException("Cannot write to a read-only property $class::\$$name.");
+
+		} else {
+			ObjectMixin::strictSet($class, $name);
+		}
+	}
+
+
+	/**
+	 * @return void
+	 * @throws MemberAccessException
+	 */
+	public function __unset($name)
+	{
+		$class = get_class($this);
+		if (!ObjectMixin::hasProperty($class, $name)) {
+			throw new MemberAccessException("Cannot unset the property $class::\$$name.");
+		}
+	}
+
+
+	/**
+	 * @return bool
+	 */
+	public function __isset($name)
+	{
+		$uname = ucfirst($name);
+		return ObjectMixin::getMagicProperty(get_class($this), $name)
+			|| ($name !== '' && ($methods = ObjectMixin::getMethods(get_class($this))) && (isset($methods['get' . $uname]) || isset($methods['is' . $uname])));
+	}
+
+
+	/**
+	 * @return Reflection\ClassType|\ReflectionClass
+	 * @deprecated
+	 */
+	public static function getReflection()
+	{
+		trigger_error(get_called_class() . '::getReflection() is deprecated' . ObjectMixin::getSource(), E_USER_DEPRECATED);
+		$class = class_exists(Reflection\ClassType::class) ? Reflection\ClassType::class : \ReflectionClass::class;
+		return new $class(get_called_class());
+	}
+
+
+	/**
+	 * @return mixed
+	 * @deprecated use Nette\Utils\ObjectMixin::setExtensionMethod()
+	 */
+	public static function extensionMethod($name, $callback = NULL)
+	{
+		if (strpos($name, '::') === FALSE) {
+			$class = get_called_class();
+		} else {
+			list($class, $name) = explode('::', $name);
+			$class = (new \ReflectionClass($class))->getName();
+		}
+		trigger_error("Extension methods like $class::$name() are deprecated" . ObjectMixin::getSource(), E_USER_DEPRECATED);
```

Comment:
I think this should say "Extension methods such as" instead of "Extension methods like".